llvm/lib/Target/RISCV/RISCVInsertVSETVLI.cpp

```diff
@@ -1161,10 +1161,7 @@ void RISCVInsertVSETVLI::insertVSETVLI(MachineBasicBlock &MBB,
     // isn't always the case, e.g. PseudoVMV_X_S doesn't have an AVL operand or
     // we've taken the AVL from the VL output of another vsetvli.
     LiveInterval &LI = LIS->getInterval(AVLReg);
-    // Need to get non-const VNInfo
-    VNInfo *VNI = LI.getValNumInfo(Info.getAVLVNInfo()->id);
-    LI.addSegment(LiveInterval::Segment(
-        VNI->def, LIS->getInstructionIndex(*MI).getRegSlot(), VNI));
+    LIS->extendToIndices(LI, {LIS->getInstructionIndex(*MI).getRegSlot()});
```

Comment:
Why do we need to call `getRegSlot` again? Does `getInstructionIndex` return the `SlotIndex`?

Reply:
I think `getInstructionIndex` returns the `Slot_Block` SlotIndex, e.g. 80B. `getRegSlot()` turns it into 80r which is the time when the operands are defined/used.